bagel/tests/test_cli_bids.py

```diff
@@ -57,6 +58,39 @@ def test_bids_sessions_have_correct_labels(
         ]
 
 
+def test_bids_sessions_have_correct_paths(
+    runner,
+    test_data,
+    tmp_path,
+    load_test_json,
+):
+    """
+    Check that BIDS session paths added to pheno_bids.jsonld match the parent
+    session/subject labels and are absolute file paths.
+    """
+    runner.invoke(
+        bagel,
+        [
+            "bids",
+            "--jsonld-path",
+            test_data / "example_synthetic.jsonld",
+            "--bids-dir",
+            Path(__file__).parent.parent.parent / "bids-examples/synthetic",
+            "--output",
+            tmp_path,
+        ],
+    )
+
+    pheno_bids = load_test_json(tmp_path / "pheno_bids.jsonld")
+    for sub in pheno_bids["hasSamples"]:
+        for ses in sub["hasSession"]:
+            assert all(
+                label in ses["filePath"]
```

Comment:
I think two separate assert statements make more sense here. 1) It is going to be more readable, and 2) it will be more readable in the test summary if it fails.

If I understand right, you want to see that the subject and / or session name will show up as part of the filePath attribute? Would this test then fail for a dataset that doesn't have a session layer (i.e. where we have created a session on our own). If so, I think then we should have a separate test for "bids_had_session" and "bids_didnt_have_session" (better names ideally) to make this very explicit.

Reply:
Agree with the two assert statements!

You're right that if the dataset used for this test were one that didn't have a session layer (as opposed to `example_synthetic`), this test would fail - though my understanding is that that is currently an edge case which we may choose to handle differently in the future (recalling your question of whether we should create a session directory so that the `pheno_bids.ld` matches what's on disk). 

Can definitely rename this test to be more explicit in terms of which case it covers. Since this test currently depends on the CLI runner though, I can't easily write a `bids_didnt_have_session` version since we don't currently have a matching example pheno.json and participants.tsv for any of the relevant BIDS examples.

One thing I could do is factor out the code that gets the paths (below) into its own function and test it with a unit test instead. I didn't do this before since it's only one line of code anyways per condition (and theoretically, I would just be passing in arguments for `layout.get()` ). What do you think?

https://github.com/neurobagel/bagel-cli/blob/2e139c9894483febc30df911180c6c43f0608871/bagel/cli.py#L253-L277